src/router.js

```diff
@@ -33,5 +38,46 @@ export default [
 	{ path: '/tv/:id', name: 'tvShowDetails', component: TvShowDetails },
 	{ path: '/tv/:id/seasons', name: 'seasons', component: Seasons },
 	{ path: '/tv/:id/seasons/:number', name: 'episodes', component: Episodes },
-	{ path: '/person/:id', name: 'actorDetails', component: ActorDetails }
+	{ path: '/person/:id', name: 'actorDetails', component: ActorDetails },
+	{
+		path: '/login',
+		name: 'login',
+		component: Login,
+		meta: {
+			requiresAuth: true
+		}
+	},
+	{
+		path: '/profile',
+		component: Profile,
+		meta: {
+			requiresProfile: true
+		}
+	}
 ];
+
+Vue.use(VueRouter);
+
+const router = new VueRouter({
+	routes,
+	mode: 'history'
+});
+
+router.beforeEach((to, from, next) => {
+	if (to.matched.some((record) => record.meta.requiresAuth)) {
+		if (!store.getters.user) {
+			next();
+			return;
+		}
+		next('/profile');
```

Comment:
```suggestion
		next();
```
Please have a look how navigation guards work https://router.vuejs.org/guide/advanced/navigation-guards.html